infrastructure/file_settings_source.go

```diff
@@ -0,0 +1,62 @@
+package infrastructure
+
+import (
+	"encoding/json"
+
+	bosherr "github.com/cloudfoundry/bosh-agent/errors"
+	boshlog "github.com/cloudfoundry/bosh-agent/logger"
+	boshsettings "github.com/cloudfoundry/bosh-agent/settings"
+	boshsys "github.com/cloudfoundry/bosh-agent/system"
+)
+
+type FileSettingsSource struct {
+	metaDataFilePath string
+	userDataFilePath string
+	settingsFilePath string
+
+	fs boshsys.FileSystem
+
+	logger boshlog.Logger
+	logTag string
+}
+
+func NewFileSettingsSource(
+	metaDataFilePath string,
+	userDataFilePath string,
```

Comment:
you can remove `metaDataFilePath` and `userDataFilePath` from this class since it only uses settingsFilePath.